src/java.desktop/macosx/classes/sun/lwawt/macosx/CPlatformWindow.java

```diff
@@ -986,13 +986,15 @@ public void setWindowState(int windowState) {
                     // let's return into the normal states first
                     // the zoom call toggles between the normal and the max states
                     unmaximize();
+                    peer.displayChanged();
```

Comment:
Why do you call `peer.displayChanged()` only when the previous windows state is `MAXIMIZED_BOTH`? What happens if the previous windows state is `NORMAL`?